compiler/src/dotty/tools/dotc/transform/CompleteJavaEnums.scala

```diff
@@ -144,7 +149,12 @@ class CompleteJavaEnums extends MiniPhase with InfoTransformer { thisPhase =>
         templ.body.collect {
           case mdef: DefDef if mdef.name == name => mdef.rhs
         }.head
-      val args = List(rhsOf(nme.toString_), rhsOf(nme.ordinalDollar))
+      def nameArg =
+        if cls.owner.name eq nme.DOLLAR_NEW then // productPrefix calls .name so we need to find the argument from $new
+          ref(cls.owner.paramSymss.flatten.find(p => p.isTerm && p.name == nme.nameDollar).map(_.termRef).get)
```

Comment:
Is `find` really needed to find this as opposed to doing `.member(nme.nameDollar)` on the class or something?

Reply:
It doesn't really look like member works for methods, but its possible to just do `ref(cls.owner.paramSymss.head.tail.head)`

Reply:
ah, cause it's only a method parameter right, so i guess a search is fine, but i would do `.head.find` instead of `.flatten.find`

Reply:
I managed to simplify to `ref(cls.owner.paramSymss.head.find(_.name == nme.nameDollar).get)`